pkgs/test/nixpkgs-check-by-name/src/references.rs

```diff
@@ -1,17 +1,17 @@
-use crate::structure::Nixpkgs;
+use crate::check_result;
+use crate::check_result::CheckResult;
+use crate::nixpkgs_problem::NixpkgsProblem;
 use crate::utils;
-use crate::utils::{ErrorWriter, LineIndex};
+use crate::utils::LineIndex;
 
 use anyhow::Context;
 use rnix::{Root, SyntaxKind::NODE_PATH};
 use std::ffi::OsStr;
 use std::fs::read_to_string;
-use std::io;
 use std::path::{Path, PathBuf};
 
 /// Small helper so we don't need to pass in the same arguments to all functions
-struct PackageContext<'a, W: io::Write> {
-    error_writer: &'a mut ErrorWriter<W>,
+struct PackageContext<'a> {
     /// The package directory relative to Nixpkgs, such as `pkgs/by-name/fo/foo`
     relative_package_dir: &'a PathBuf,
     /// The absolute package directory
```

Comment:
You didn't touch this in this PR, but it causes the new code below to be somewhat needlessly verbose:

It is almost always a bad idea to have `&` to the owned versions of sequences, e.g. `&Vec<T>`, `&String`, `&PathBuf`.
You don't get any extra powers over just using the borrowed version, i.e. `&[T]`, `&str`, `&Path`, because you still can't mutate/move the thing. You still need to drag that lifetime parameter around, and you have a double indirection (in terms of pointers) when a single one would've sufficed.
And as you see in the code below, you now need some extra verbosity to turn `&Path` into `&PathBuf` at various points.

I'd suggest restricting yourself to using only `PathBuf` and `&'a Path`.